devdoc/channel.md

```diff
@@ -137,7 +137,7 @@ MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_push, THANDLE(CHANNEL), channel, THA
 
 ### channel_pull
 ```c
-    MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_pull, THANDLE(CHANNEL), channel, PULL_CALLBACK, pull_callback, void*, pull_context, THANDLE(ASYNC_OP)*, out_op_pull);
+    MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_pull, THANDLE(CHANNEL), channel, THANDLE(RC_STRING), correlation_id, PULL_CALLBACK, pull_callback, void*, pull_context, THANDLE(ASYNC_OP)*, out_op_pull);
```

Comment:
>correlation_id [](http://example.com/codeflow?start=101&length=14)

What happens when `correlation_id` is "AAA" and `THANDLE(PTR(LOG_CONTEXT_HANDLE), log_context` is `NULL`?

Reply:
There is a log line on line 154 in `channel_internal.c`. 

That log line prints the log_context and correlation_id. The correlation_ids are passed to the callbacks as well. 

If `log_context` is `NULL` but `correlation_id` is a valid string, they will just be printed as such.